internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -223,6 +242,11 @@ func (r *Reconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl.Result
 		return ctrl.Result{}, errors.Wrap(err, "error reconciling the Cluster topology")
 	}
 
+	requeueAfter := s.HookResponseTracker.EffectiveRetryAfter()
```

Comment:
Could we add a comment here why we expect that `requeueAfter` may be `!= 0`?

```suggestion
	// requeueAfter will not be 0 if a runtime SDK hook failed and failurePolicy is fail.
	requeueAfter := s.HookResponseTracker.EffectiveRetryAfter()
```

Reply:
Let's please address this nit in Yuvaraj's next PR if there are no other findings. I don't want to defer merging this PR by almost a day (the PR for the next hook becomes way simpler if we can already merge this PR + rebase)

Reply:
ok 4 me

Reply:
Ah just remembered I can also push the fix on this PR. Let's wait for Fabrizio's review